lib/editor/components/pattern/PatternStopContainer.js

```diff
@@ -157,12 +159,18 @@ class PatternStopContainer extends Component<Props, State> {
           }
           // Since stops and locations are unified, the stop could also refer to a location
           // IDs exist in one namespace as per the spac
-          if (!stop) {
+          if (card.hasOwnProperty('locationId') && !card.hasOwnProperty('locationGroupId')) {
```

Comment:
Given the number of times these checks are made, I'm starting to think we need a some util function that makes the distinction (or a function that returns an object of the correct type so you don't have to write all of these flow waivers).

Reply:
There is already `patternHaltIsLocation` and `patternHaltIsLocationGroup` which I've now used here.